ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -62,6 +62,14 @@ router bgp {{ router_bgp.as }}
 {%         if router_bgp.peer_groups[peer_group].update_source is arista.avd.defined %}
    neighbor {{ peer_group }} update-source {{ router_bgp.peer_groups[peer_group].update_source }}
 {%         endif %}
+{%         if router_bgp.peer_groups[peer_group].rib_in_pre_policy_retain.all is arista.avd.defined(true) %}
```

Comment:
- I am not sure how this command behaves in eos with the combination of `no neighbor .... retain all`. Would you still need to add the `all` if it is `no ... ` ?
- You are not printing out the command.
- I would prefer if you could swap the order, so you just add the ` all` if needed at before printing.

Reply:
@ClausHolbechArista it is interesting, per the [docs ](https://www.arista.com/en/um-eos/eos-border-gateway-protocol-bgp?searchword=neighbor%20rib-in%20pre-policy%20retain) the following syntax is documented:

```
neighbor neighbor_ID rib-in pre-policy retain [all]

no neighbor neighbor_ID rib-in pre-policy retain

default neighbor neighbor_ID rib-in pre-policy retain
```

But on my EOS 4.25.4M, I see the following:
```
neighbor neighbor_ID rib-in pre-policy retain [all]

neighbor neighbor_ID rib-in pre-policy retain [disabled]

no neighbor neighbor_ID rib-in pre-policy retain

default neighbor neighbor_ID rib-in pre-policy retain
```

Reply:
If we leave the discussion on `disabled` aside (you have another issue open for that), this means that `all` should only be printed in the `enabled case`.
I would suggest:
```yaml
router_bgp:
  peer_groups:
     test:
       rib_in_pre_policy_retain:
         enabled: < true | false >
         all: < true | false >
```
And then only evaluate the `all` field in the `enabled: true` case.

Reply:
Thanks @ClausHolbechArista I will continue down this path.

Reply:
ok to close the loop here
defaulting the command sets the following: `neighbor neighbor_ID rib-in pre-policy retain`

So in reality the logic would be `rib_in_pre_policy_retain['enabled'] == true` OR `rib_in_pre_policy_retain not defined`
would generate no config which is the same as `neighbor neighbor_ID rib-in pre-policy retain` since its the default config.

Next the command if explicitly set, can only lead to `all` or `disabled`, and both keys cannot be set.

Given the above I think the proper model should be:
```
router_bgp:
  peer_groups:
     test:
       rib_in_pre_policy_retain: < true | false or disabled | all >
```

BUT that seems sort of confusing.  Thoughts @ClausHolbechArista ?

We can chat on the Arista slack if it is easier.

Reply:
As discussed with @ClausHolbechArista I will skip the disabled config syntax given it does not fit well into the current framework because it requires logic in the template or input validation (which does not exist).  I will use the following dictionary with this in mind:

```
router_bgp:
  peer_groups:
     test:
       rib_in_pre_policy_retain:
         enabled: < true | false >
         all: < true | false >
```